src/esp/scene/ReplicaSemanticScene.cpp

```diff
@@ -102,25 +103,27 @@ bool SemanticScene::buildReplicaHouse(const io::JsonDocument& jsonDoc,
     }
 
     const auto& obb = jsonObject["oriented_bbox"];
-    const vec3f aabbCenter = io::jsonToVec3f(obb["abb"]["center"]);
-    const vec3f aabbSizes = io::jsonToVec3f(obb["abb"]["sizes"]);
-
-    const vec3f translationBoxToWorld =
-        io::jsonToVec3f(obb["orientation"]["translation"]);
-
-    std::vector<float> rotationBoxToWorldCoeffs;
-    io::toFloatVector(obb["orientation"]["rotation"],
-                      &rotationBoxToWorldCoeffs);
-    const Eigen::Map<quatf> rotationBoxToWorld(rotationBoxToWorldCoeffs.data());
-
-    Eigen::Isometry3f transformBox{rotationBoxToWorld.normalized()};
-    transformBox *= Eigen::Translation3f(translationBoxToWorld);
-
-    const Eigen::Isometry3f transformBoxToWorld{worldRotationMat *
-                                                transformBox};
-
-    object->obb_ = geo::OBB{transformBoxToWorld * aabbCenter, aabbSizes,
-                            quatf{transformBoxToWorld.linear()}.normalized()};
+    Mn::Vector3 aabbCenter;
+    io::fromJsonValue(obb["abb"]["center"], aabbCenter);
+    Mn::Vector3 aabbSizes;
+    io::fromJsonValue(obb["abb"]["sizes"], aabbSizes);
+    Mn::Vector3 translationBoxToWorld;
+    io::fromJsonValue(obb["orientation"]["translation"], translationBoxToWorld);
+    // 4th element is scalar in json
+    Mn::Vector4 rotBoxToWorldCoeffs;
+    io::fromJsonValue(obb["orientation"]["rotation"], rotBoxToWorldCoeffs);
```

Comment:
A drive-by fix could be to not do repeated JSON key lookups for `"aabb"` and `"orientation"`.

Also, can't the `fromJsonValue` populate a quat already? Since you're subsequently making a quat from it anyway.